File: src/jvmTest/java/LongLongHashMapTestGenerator.java
import com.google.common.collect.testing.Helpers;
import com.google.common.collect.testing.SampleElements;
import com.google.common.collect.testing.TestMapGenerator;
import example.LongLongLinkedHashMap;
import org.jetbrains.annotations.NotNull;

import java.util.List;
import java.util.Map;

@SuppressWarnings("unchecked")
public class LongLongHashMapTestGenerator implements TestMapGenerator<Long, Long> {
    public Long[] createKeyArray(final int length) {
        return new Long[length];
    }

    public Long[] createValueArray(final int length) {
        return new Long[length];
    }

    public SampleElements<Map.Entry<Long, Long>> samples() {
        return new SampleElements<>(
                Helpers.mapEntry(1L, 123L),
                Helpers.mapEntry(2L, 234L),
                Helpers.mapEntry(3L, 345L),
                Helpers.mapEntry(345L, 6L),
                Helpers.mapEntry(777L, 666L));
    }

    public Map<Long, Long> create(@NotNull final Object... entries) {
        Map<Long, Long> map = (Map<Long, Long>) new LongLongLinkedHashMap();
        for (Object o : entries) {
            Map.Entry<Long, Long> entry = (Map.Entry<Long, Long>) o;
            map.put(entry.getKey(), entry.getValue());
        }
        return map;
    }

    public Map.Entry<Long, Long>[] createArray(final int length) {
        return new Map.Entry[length];
    }

    public Iterable<Map.Entry<Long, Long>> order(final List<Map.Entry<Long, Long>> insertionOrder) {
        return insertionOrder;
    }
}

File: src/jvmTest/kotlin/LongLongHashMapTestGenerator.kt
import com.google.common.collect.testing.Helpers
import com.google.common.collect.testing.SampleElements
import com.google.common.collect.testing.TestMapGenerator
import example.createLinkedOpenHashMap

class LongLongHashMapTestGenerator : TestMapGenerator<Long?, Long?> {
    override fun createKeyArray(length: Int) = arrayOfNulls<Long?>(length)
    override fun createValueArray(length: Int) = arrayOfNulls<Long?>(length)
    override fun createArray(length: Int) = arrayOfNulls<Map.Entry<Long?, Long?>>(length)
    override fun order(insertionOrder: List<Map.Entry<Long?, Long?>>) = insertionOrder

    override fun samples(): SampleElements<Map.Entry<Long?, Long?>> {
        return SampleElements(
            Helpers.mapEntry(1L, 123L),
            Helpers.mapEntry(2L, 234L),
            Helpers.mapEntry(3L, 345L),
            Helpers.mapEntry(345L, 6L),
            Helpers.mapEntry(777L, 666L)
        )
    }

    override fun create(vararg entries: Any): Map<Long?, Long?> {
        val map = createLinkedOpenHashMap<Long, Long>() as MutableMap<Long?, Long?>
        for (o in entries) {
            val entry = o as Map.Entry<Long?, Long?>?
            map[entry!!.key] = entry.value
        }
        return map
    }
}